src/main/java/com/redhat/pantheon/sling/PantheonRepositoryInitializer.java

```diff
@@ -36,13 +38,36 @@
 
     private static final Logger log = LoggerFactory.getLogger(PantheonRepositoryInitializer.class);
 
+    private ServiceResourceResolverProvider serviceResourceResolverProvider;
+
+    @Activate
+    public PantheonRepositoryInitializer(@Reference ServiceResourceResolverProvider serviceResourceResolverProvider) {
+        this.serviceResourceResolverProvider = serviceResourceResolverProvider;
+    }
+
     @Override
     public void processRepository(SlingRepository slingRepository) throws Exception {
-        initializeRepositoryACLs(slingRepository);
+        initializeRepositoryACLs(getSession(slingRepository));
+        setGitServiceURL(getSession(slingRepository));
+    }
+
+    private JackrabbitSession getSession(SlingRepository slingRepository) throws RepositoryException {
+        return (JackrabbitSession) slingRepository.loginAdministrative(null);
+    }
+
+    private void setGitServiceURL(JackrabbitSession s) throws RepositoryException {
+        if (System.getenv("GIT_SERVICE_URL") != null) {
+            assignPermissionToPrincipal(s,"pantheon","/conf/pantheon", null, Privilege.JCR_READ, Privilege.JCR_WRITE);
+            assignPermissionToPrincipal(s,"pantheon-users","/conf/pantheon", null, Privilege.JCR_READ);
+            s.getNode("/conf/pantheon").setProperty("pant:GitServiceURL",System.getenv("GIT_SERVICE_URL"));
```

Comment:
Is this `GIT_SERVICE_URL` property just in case you're hosting your code somewhere other than github, so you can link to it correctly?

Edit: No, that must not be it, since github.com is hardcoded in the link redirect. So what is the purpose of this code?

Reply:
This is the git2pantheon URL.